owl/EASE-IO.owl

```diff
@@ -51,6 +62,69 @@
         <rdfs:comment>An information realization with object character. e.g. a software agent etc.</rdfs:comment>
         <rdfs:label>Digital object</rdfs:label>
     </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/EASE-IO.owl#FaultySoftware -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#FaultySoftware">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/EASE.owl#SoftwareDiagnosis"/>
+        <rdfs:comment>A description of a situation where some software has a bug.</rdfs:comment>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/EASE-IO.owl#IncompatibleSoftware -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#IncompatibleSoftware">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/EASE.owl#SoftwareDiagnosis"/>
+        <rdfs:comment>A description of a situation where two software systems are incompatible with each other.</rdfs:comment>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/EASE-IO.owl#SoftwareImplementation -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#SoftwareImplementation">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/EASE-IO.owl#DigitalObject"/>
+        <rdfs:comment>An information realization in form of a executable computer program.</rdfs:comment>
+        <rdfs:label>Software implementation</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/EASE-IO.owl#SoftwarePackage -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#SoftwarePackage">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/EASE-IO.owl#DigitalObject"/>
+        <rdfs:comment>A structured archive of software that also often contains data needed to run the software, and scripts to build and configure the software.</rdfs:comment>
+        <rdfs:label>Software package</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/EASE-IO.owl#SoftwareType -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#SoftwareType">
```

Comment:
This is very nice!

Reply:
Still hard to come up with a good taxonomy! I tried already, the current result is in the *SRDL* package of KnowRob. But I think this will need a couple of iterations to only cover software of robot control systems.